### PR TITLE
multi: synchronous rpc middleware registration

### PR DESCRIPTION
**Pull request by ellemouton**
In this commit, we change the flow of the rpc middleware registration
a bit. In order to allow a client to add rpc middleware interceptors in
a deterministic order, we now make the server send a "registration
complete" message to the client after completing the registration
process so that the client knows when it can go ahead and register the
next client.

### CODE REVIEW

**Review by guggero — APPROVED** [review]
LGTM :tada:

File: lnrpc/lightning.proto [hunk 1]
```diff
@@ -4379,6 +4379,14 @@ message RPCMiddlewareRequest {
         the same type, or replaced by an error message.
         */
         RPCMessage response = 6;
+
+        /*
+        This is used to indicate to the client that the server has successfully
```
Comment (yyforyongyu) [inline lnrpc/lightning.proto:4384]: So this is like a server ack?

**Review by yyforyongyu — APPROVED** [review]
LGTM💡

File: lnrpc/lightning.proto (hunk 1 above)
Comment (ellemouton) [reply lnrpc/lightning.proto:4384]: yeah exactly :)

**Pull request merged**